### PR TITLE
fix: [CDS-77820]: Close InputSetSelector modal on click of Apply for trigger flow

### PR DESCRIPTION
**Pull request by pankajsingh-harness**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by Eric761 — APPROVED** [review]

**Review by sonaligoyal-harness — APPROVED** [review]

File: src/modules/70-pipeline/components/InputSetSelector/InputSetSelector.tsx [hunk 1]
```diff
@@ -414,7 +414,9 @@ export function InputSetSelector({
                     disabled={!selectedInputSets?.length || anyInputSetInLoadingState}
                     onClick={() => {
                       onChange?.(selectedInputSets)
-                      if (reRunInputSetYaml || isSimplifiedYAML) setOpenInputSetsList(false)
+                      if (reRunInputSetYaml || closeInputSetModalOnApply) {
+                        setOpenInputSetsList(false)
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/components/InputSetSelector/InputSetSelector.tsx:418]: We can close modal in every cases. Can we confirm this with @sunnykesh-harness and make it uniform ?

**Review by chetanvekariya-harness — APPROVED** [review]

**Comment by pankajsingh-harness** [comment]
trigger featurenamecheck

**Comment by pankajsingh-harness** [comment]
retrigger eslint

File: src/modules/70-pipeline/components/InputSetSelector/InputSetSelector.tsx (hunk 1 above)
Comment (pankajsingh-harness) [reply src/modules/70-pipeline/components/InputSetSelector/InputSetSelector.tsx:418]: @chetanvekariya-harness I was also thinking the same. Connected with @Eric761 on this. He only added this to handle one specific use case related to reconcile.

**Comment by pankajsingh-harness** [comment]
retrigger eslint

**Pull request merged**